java/maven.hints/src/org/netbeans/modules/maven/hints/errors/EnablePreviewMavenProj.java

```diff
@@ -49,148 +43,146 @@
+            return new ActionConfig(actionName, "exec.args");
         }
+        public static ActionConfig testAction(String actionName) {
+            return new ActionConfig(actionName, "argLine");
+        }
+    }
 
-        @Override
-        public ChangeInfo implement() throws Exception {
-
-            try {
-
-                final FileObject pom = prj.getProjectDirectory().getFileObject("pom.xml"); // NOI18N
-                pom.getFileSystem().runAtomicAction(new FileSystem.AtomicAction() {
-                    @Override
-                    public void run() throws IOException {
-                        List<ModelOperation<POMModel>> operations = new ArrayList<ModelOperation<POMModel>>();
-                        operations.add(new AddMvnCompilerPluginForEnablePreview());
-                        org.netbeans.modules.maven.model.Utilities.performPOMModelOperations(pom, operations);
+    private static class BaseMvnCompilerPluginForEnablePreview {
+
+        protected static final String MAVEN_COMPILER_GROUP_ID = "org.apache.maven.plugins"; // NOI18N
+        protected static final String MAVEN_COMPILER_ARTIFACT_ID = "maven-compiler-plugin"; // NOI18N
+        protected static final String COMPILER_ID_PROPERTY = "compilerId"; // NOI18N
+        protected static final String RELEASE = "release"; // NOI18N
+        protected static final String RELEASE_PROPERTY = "maven.compiler.release"; // NOI18N
+        protected static final String SOURCE = "source"; // NOI18N
+        protected static final String SOURCE_PROPERTY = "maven.compiler.source"; // NOI18N
+        protected static final String TARGET = "target"; // NOI18N
+        protected static final String TARGET_PROPERTY = "maven.compiler.target"; // NOI18N
+        protected static final String COMPILER_ARG = "compilerArgs"; // NOI18N
+        protected static final String MAVEN_COMPILER_VERSION = "3.11.0"; // NOI18N
+        protected static final String ARG = "arg";// NOI18N
+
+        protected Plugin searchMavenCompilerPlugin(final Build build) {
+            List<Plugin> plugins = build.getPlugins();
+            if (plugins != null) {
+                for (Plugin plugin : plugins) {
+                    if (MAVEN_COMPILER_GROUP_ID.equals(plugin.getGroupId())
+                            && MAVEN_COMPILER_ARTIFACT_ID.equals(plugin.getArtifactId())) {
+                        return plugin;
                     }
-                });
-
-            } catch (IOException ex) {
+                }
             }
-            ProjectConfiguration cfg = prj.getLookup().lookup(ProjectConfigurationProvider.class).getActiveConfiguration();
-
-            for (String action : new String[]{"run", "debug", "profile"}) { // NOI18N
-
-                NetbeansActionMapping mapp = ModelHandle2.getMapping(action, prj, cfg);
-                Map<String, String> properties = mapp.getProperties();
+            return null;
+        }
```

Comment:
my mistake, I didn't see that this method was implemented in the super class.

I think it would be better to move the method from `AddMvnCompilerPluginForEnablePreview` to `BaseMvnCompilerPluginForEnablePreview`, so that it is not implemented twice, just slightly differently

Reply:
Done.

Reply:
awesome, thanks!